tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs

```diff
@@ -63,7 +63,37 @@ public async Task ProcessRunAsync(Uri runUri)
             var projectClient = vssConnection.GetClient<ProjectHttpClient>();
 
             var project = await projectClient.GetProject(projectGuid.ToString());
-            var build = await buildClient.GetBuildAsync(projectGuid, runId);
+
+            // Avoid unhandled BuildNotFoundException by folding it into the following null check.
+            Build build = null;
+
+            try
+            {
+                build = await buildClient.GetBuildAsync(projectGuid, runId);
+            }
+            catch (BuildNotFoundException)
+            {
+            }
+
+            if (build == null)
+            {
+                this.logger.LogWarning("Unable to process run due to missing build");
```

Comment:
In the case that this isn't already in the log context.

```suggestion
                this.logger.LogWarning($"Unable to process run due to missing build for run {runId}");
```